tests/prague/eip2935_historical_block_hashes_from_state/test_block_hashes.py

```diff
@@ -190,7 +196,48 @@ def test_block_hashes_history_at_transition(
     post[Address(current_code_address)] = Account(storage=storage)
     current_code_address += 0x100
 
-    blocks.append(Block(timestamp=FORK_TIMESTAMP, txs=txs))
+    blocks.append(Block(timestamp=Spec.FORK_TIMESTAMP, txs=txs))
+
+    # Add blocks after the fork transition to gradually fill up the `HISTORY_SERVE_WINDOW`
+    for i in range(1, blocks_after_fork + 1):
+        current_block_number += 1
+        txs = []
+        # On these blocks, `BLOCKHASH` will still return values for the last 256 blocks, and
+        # `HISTORY_STORAGE_ADDRESS` should now serve values for the previous blocks in the new
+        # fork.
+        code = b""
+        storage = Storage()
+
+        # Check that each block can return previous blockhashes within `BLOCKHASH_OLD_WINDOW``
+        for j in range(1, min(i, Spec.BLOCKHASH_OLD_WINDOW) + 1):
+            code += generate_block_check_code(
+                block_number=(current_block_number - j),
+                populated_blockhash=True,
+                populated_history_storage_contract=True,
+                storage=storage,
+            )
+
+        # Check that each block can return previous blockhashes within `HISTORY_SERVE_WINDOW`
+        for j in range(Spec.BLOCKHASH_OLD_WINDOW + 1, min(i, Spec.HISTORY_SERVE_WINDOW) + 1):
+            code += generate_block_check_code(
+                block_number=(current_block_number - j),
+                populated_blockhash=False,
```

Comment:
We could merge this loop with the one above and calculate `populated_blockhash` dending on the block number.

Also we need to check outside of the `HISTORY_SERVE_WINDOW` range when possible.